src/twisted/web/test/_util.py

```diff
@@ -13,10 +13,13 @@
 from twisted.web import server
 from twisted.web._flatten import flattenString
 from twisted.web.error import FlattenerError
+from twisted.web.http import Request
+from twisted.web.resource import IResource
 from twisted.web.template import Flattenable
+from .requesthelper import DummyRequest
 
 
-def _render(resource, request):
+def _render(resource: IResource, request: Request | DummyRequest) -> Deferred[None]:
```

Comment:
should there be an Interface that both http.Request and DummyRequest implement? Currently there's no interface with `notifyFinish()`

Reply:
the HTTP interfaces are not the best .

I think the current annotation is ok. thanks

What I did, I have created my own DummyRequest that  inherits from http.Request and does the minimum overrwriting possible.